src/fast_api.rs

```diff
@@ -267,13 +267,13 @@ impl<T: Default> FastApiTypedArray<T> {
   /// Returns a slice pointing to the underlying data if safe to do so.
   #[inline(always)]
   pub fn get_storage_if_aligned(&self) -> Option<&mut [T]> {
-    // Ensure that we never create a null-ptr slice (even a zero-length null-ptr slice
-    // is invalid because of Rust's niche packing).
-    if self.data.is_null() {
+    // V8 may provide an invalid or null pointer when length is zero, so we just
```

Comment:
**note:** V8 returning an invalid pointer is a bug, I've reported it to them some time ago and there was some attempts at fixing it but it wasn't completely trivial.

The basic reason why it wasn't trivial was because for zero-length buffers the backing pointer can either be null or non-null and they ran into some complexities with that.